src/Evolution/Systems/ScalarWave/Constraints.hpp

```diff
@@ -0,0 +1,107 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/ScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+
+/// \cond
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace ScalarWave {
+// @{
+/*!
+ * \brief Compute the scalar-wave one-index constraint.
+ *
+ * \details Computes the scalar-wave one-index constraint,
+ * \f$C_{i} = \partial_i\psi - \Phi_{i},\f$ which is
+ * given by Eq. (19) of \cite Holst2004wt
+ */
+template <size_t SpatialDim>
+tnsr::i<DataVector, SpatialDim, Frame::Inertial> one_index_constraint(
+    const tnsr::i<DataVector, SpatialDim, Frame::Inertial>& d_psi,
+    const tnsr::i<DataVector, SpatialDim, Frame::Inertial>& phi) noexcept;
+
+template <size_t SpatialDim>
+void one_index_constraint(
+    gsl::not_null<tnsr::i<DataVector, SpatialDim, Frame::Inertial>*> constraint,
+    const tnsr::i<DataVector, SpatialDim, Frame::Inertial>& d_psi,
+    const tnsr::i<DataVector, SpatialDim, Frame::Inertial>& phi) noexcept;
+// @}
+
+// @{
+/*!
+ * \brief Compute the scalar-wave 2-index constraint.
+ *
+ * \details Computes the scalar-wave 2-index constraint
+ * \f$C_{ij} = \partial_i\Phi_j - \partial_j\Phi_i,\f$
+ * where \f$\Phi_{i} = \partial_i\psi\f$, that is given by
+ * Eq. (20) of \cite Holst2004wt
+ *
+ * \note We do not native support custom storage for antisymmetric tensors yet.
```

Comment:
This doesn't read quite right - I suggest "We do not support..."

Reply:
fixed, thanks!